### ISSUE TITLE
Loss of data when reading from .laz

### ISSUE THREAD

**Issue by anitagraser** [qgis/QGIS#45307]
### What is the bug or the crash?

There may be an issue with loss of data when reading from laz files. 
In particular, the files published by the city of Vienna: https://www.wien.gv.at/gogv/l9ogdkappazunderdatenpaket

As described in https://anitagraser.com/2021/09/25/exploring-viennas-street-level-lidar-kappazunder-data-sample/, only very few points are displayed once the files are loaded. A reader of the post confirmed that "After fully converting them, there seems to be quite a loss of data, the ept-files are much to small to contain all points, and also none of the attributes are shown in QGIS. Opening the laz-Files in other software (e.g. CloudCompare) shows nice, RGB-colored laser scans. "

![image](https://user-images.githubusercontent.com/590385/135164499-281cac6f-a99c-4725-b6c8-67c7b4d500e3.png)

### Steps to reproduce the issue

Download the laz files from https://www.wien.gv.at/gogv/l9ogdkappazunderdatenpaket and try to load them.

### Versions

QGIS version
3.20.3-Odense
QGIS code revision
495fbaec
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.3.1
PROJ version
8.1.1
EPSG Registry database version
v10.028 (2021-07-07)
GEOS version
3.9.1-CAPI-1.14.2
SQLite version
3.35.2
PDAL version
2.3.0
PostgreSQL client version
13.0
SpatiaLite version
5.0.1
QWT version
6.1.3
QScintilla2 version
2.11.5
OS version
Windows 10 Version 2009

Active Python plugins
DataPlotly
DissolveWithStats
EasyCustomLabeling
GeoCoding
Hqgis
IPyConsole
kmltools
LAStools
MemoryLayerSaver
mmqgis
networks
Offline-MapMatching
ORStools
pdaltools
pluginbuilder3
PluginLoadTimes
plugin_reloader
processing_edgebundling
processing_r
qconsolidate3
Qgis2threejs
qgis2web
qgis_resource_sharing
QuickOSM
QuickWKT
quick_map_services
SRTM-Downloader
db_manager
MetaSearch
processing

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by wonder-sk** [comment]
Upstream error in untwine: https://github.com/hobu/untwine/issues/75

A related upstream issue is that the errors from Untwine are not passed to QGIS, so the user does not know what went wrong: https://github.com/qgis/QGIS/issues/41765

As a workaround, one could use this kind of PDAL pipeline to just drop scan angle values. Create a new file `fix-scan-angle.json` with the following content:
```
[
    {
        "type":"readers.las",
        "filename":"scandata_162.laz"
    },
    {
        "type":"filters.assign",
        "assignment":"ScanAngle[:]=0"
    },
    {
        "type":"writers.las",
        "filename":"scandata_162+.laz"
    }
]
```
And run it like this:
```
pdal pipeline fix-scan-angle.json
```

The fixed file then loads fine in QGIS:
![wien-lidar](https://user-images.githubusercontent.com/193367/135284370-b07272bb-be8a-47ac-b050-d6024613c63b.png)

**Comment by alexbruy** [comment]
Seems fixed in the latest master (8c62d89729ec9a18f2f357dcebdcd483234d11b9).
